python/cudf/cudf/core/dataframe.py

```diff
@@ -1475,7 +1475,18 @@ def reindex(
 
         if idx is not None:
             idx = idx if isinstance(idx, Index) else as_index(idx)
```

Comment:
We've since made this zero copy so no need to do the `isinstance` check here.
```suggestion
            idx = as_index(idx)
```